raster/src/main/scala/geotrellis/raster/VectorToRaster.scala

```diff
@@ -129,13 +135,16 @@ object VectorToRaster {
                 var ws = 0.0
                 val length = pts.size
 
-                cfor(0)(_ < length, _ + 1) { i =>
+                var i = 0
+                while (i < length) {
                   val point = pts(i)
                   val dX = (destX - point.geom.x)
                   val dY = (destY - point.geom.y)
                   val d = dX * dX + dY * dY
+                  i += 1
+
                   if (d < rr) {
-                    val w = 1 / d
+                    val w = if (d == 0) w0 else 1 / d
```

Comment:
My initial thought was just using the value and short-circuit as IDW definition states. This works fines and I think it is the most efficient way when you have some raster where you want to interpolate missing cells (you'll end up with a single point for each data cell).

On the other hand, there is a case were you have a set of points and you want to create an interpolation raster from them. In that case, each cell has to represent all the points that lie on its region. Thus, if we have N points which lie on the region and none has 0 distance from the center of the cell, we would assign its weighted avg to that cell. However, if only one of those points would have a 0 distance we would ignore the other points. This could be seen like contradiction, don't you think so? I'm no expert on this filed so my opinion can be biased by current use case.

About the `Infinity` problem, I don't know. But we have the same problem in case the point is not a that exact location, haven't we?.